pkg/logql/ast.go

```diff
@@ -711,12 +711,25 @@ func (e *vectorAggregationExpr) Selector() LogSelectorExpr {
 func (e *vectorAggregationExpr) Extractor() (log.SampleExtractor, error) {
 	// inject in the range vector extractor the outer groups to improve performance.
 	// This is only possible if the operation is a sum. Anything else needs all labels.
-	if r, ok := e.left.(*rangeAggregationExpr); ok && e.operation == OpTypeSum {
+	if r, ok := e.left.(*rangeAggregationExpr); ok && canInjectVectorGrouping(e.operation, r.operation) {
 		return r.extractor(e.grouping, len(e.grouping.groups) == 0)
 	}
 	return e.left.Extractor()
 }
 
+// canInjectVectorGrouping tells if a vector operation can inject grouping into the nested range vector.
+func canInjectVectorGrouping(vecOp, rangeOp string) bool {
+	if vecOp != OpTypeSum {
```

Comment:
This should work with other types as well (min, max, count), right?

Reply:
I know from fact that it doesn't work with count, we have a test in sharding that fails.

It might work for max and min but I don't have a test backing me up.

Reply:
This basically deserves a big test harness.